### PR TITLE
Read configuration from porter

### PR DESCRIPTION
**Pull request by carolynvs**
* Read config from porter on stdin
* Fallback to AZURE_STORAGE_CONNECTION_STRING

Relies on https://github.com/deislabs/porter/pull/776

### CODE REVIEW

**Comment by carolynvs** [comment]
I've rebased this after all the work we've done with vanity imports and go mods. PTAL @vdice

File: pkg/azure/blob/credentials.go [hunk 1]
```diff
@@ -1,11 +1,13 @@
-package credentials
+package blob
 
 import (
 	"os"
 	"regexp"
 
+	"get.porter.sh/plugin/azure/pkg/azure/config"
```
Comment (carolynvs) [inline pkg/azure/blob/credentials.go]: I wrote this originally in Oct and in hindsight I hate that I named this package config too. It makes it hard to tell if I'm using porter's config or this one. Do you have suggestions for differentiating the package name and/or the Config struct name?

**Comment by carolynvs** [comment]
I realize this needs tests but I really don't want to hold up this even longer and it's not being used yet. You okay with merging this if it's good and making progress on how to test a plugin in a follow-up PR?

**Comment by carolynvs** [comment]
I feel like I need a "how to test this stuff group hug" 🤗

File: pkg/azure/blob/credentials.go (hunk 1 above)
Comment (vdice) [reply pkg/azure/blob/credentials.go]: Perhaps just name the import as `azureconfig` or some such when used?  And/or `AzureConfig` as the struct?

File: pkg/azure/plugin.go [hunk 2]
```diff
@@ -17,8 +22,21 @@ func New() *Plugin {
...
+		return errors.Wrap(err, "could not read stdin")
+	}
+
+	if len(b) == 0 {
+		return nil
 	}
+
+	err = json.Unmarshal(b, &p.Config)
+	if err != nil {
+		return errors.Wrapf(err, "error unmarshaling stdin %q as azure.Config", string(b))
```
Comment (vdice) [inline pkg/azure/plugin.go]: `azure.Config` may need updating depending on what we settle on for pkg/import/struct name.  Or, perhaps we remove mention of the internal struct name itself... or generify?  (just made that one up.)

Comment (carolynvs) [reply pkg/azure/plugin.go]: Go's push to not repeat yourself (stuttering) causes more problems IMO. I'm going to try `azureconfig.Config` and see how that works.

**Pull request merged**